e2e/cli/command/test_decoder.go

```diff
@@ -0,0 +1,233 @@
+package command
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/fatih/color"
+	hclog "github.com/hashicorp/go-hclog"
+)
+
+type EventDecoder struct {
+	r io.Reader
+
+	dec    *json.Decoder
+	report *TestReport
+}
+
+type TestReport struct {
+	Events            []*TestEvent
+	Suites            map[string]*TestSuite
+	TotalSuites       int
+	TotalFailedSuites int
+	TotalCases        int
+	TotalFailedCases  int
+	TotalTests        int
+	TotalFailedTests  int
+	Elapsed           float64
+	Output            []string
+}
+
+type TestEvent struct {
```

Comment:
Should this not be defined wherever is emitting them and imported?

Reply:
unfortunately its an internal, unexported struct :man_shrugging: 
https://github.com/golang/go/blob/master/src/cmd/internal/test2json/test2json.go#L31

Reply:
lame